distribution/pull_v2.go

```diff
@@ -43,6 +47,10 @@ var (
 	errRootFSInvalid  = errors.New("invalid rootfs in image configuration")
 )
 
+const (
+	labelMediaType = "com.docker.distribution.mediaType"
```

Comment:
@tonistiigi WDYT about this label key?

Reply:
I still don't quite understand why you need it. You already call the detection function and that seems fine to me. The labeling part isn't reliable anyway for all of the cases, I'd just completely remove it.